### ISSUE TITLE
Not work on android 4.2.2 

### ISSUE THREAD

**Issue by tellfa** [renyuneyun/CamCov#8]
I want install this app on android 4.2.2 but after install and running application, when click on Enable button, not enabled and not work!!!

how can i fix this?

**Comment by renyuneyun** [comment]
Hi, do you have any apps / system settings that manage permissions?
CamCov requires camera and draw_over_other_apps (overlay) permissions.

Sorry I don't have a 4.2.2 android device in hand, so can't test it in reality.

What's your phone model? And (if you know) which ROM/system are you using? Maybe this information would help.

**Comment by tellfa** [comment]
Yes i accept any permissions for this app.
My phone models : 
HTC one x . android 4.2.2 . ROM : stock
Huawei honor 4c . android 4.2.2 . Rom : stock

please help me my bro, i really need this app and source

**Comment by tellfa** [comment]
Can you help me my bro?

**Comment by renyuneyun** [comment]
Whoops, sorry I was busy doing something else and wasn't aware of your comment.

There is only a 6.0 device in my hand right now, so I can't reproduce your situation. When using an emulator, adb complains `W/TextureView: A TextureView or a subclass can only be used with hardware acceleration enabled.` although I enabled hardware acceleration in its setting.
So I'm not sure how to help improve the situation.
Maybe, can you provide the output of adb (debugging this app) of your device?

#4 is also about Huawei honor 4c, but I'm still wondering what's the real problem since I know HTC One X uses Nvidia chips which shouldn't share the same problem (mentioned there) with MTK chips.

**Comment by renyuneyun** [comment]
My only guess is that the problem (if it's the same as #4) is due to permission checks and api versions. So I wrote a (half-)refactored version which changes the way to check permissions. Maybe you can give it a try.

It's on https://dl.bintray.com/renyuneyun/Android/ which is signed by me, so you may need to uninstall the one from fdroid (for resolving signature conflict).

**Comment by tellfa** [comment]
Hi again my friend.
I test this app on Samsung Galaxy A3 (android 5.0.2) , but again not work it! when click on Enable not show any action and not show camera!!

can you fix this bugs and update this repository??

Thanks man :heart

**Comment by renyuneyun** [comment]
Hmm, did you try the other version I uploaded to bintray (mentioned on the previous comment)? Would it work?

**Comment by tellfa** [comment]
Yes i use last version from bintray (5.0.1) but not work me again!

**Comment by renyuneyun** [comment]
That's interesting...
Do you know how to use adb and LogCat? If so, could you provide the logs (1 second or even shorter would be enough) after clicking "Enable"?
If not, I'll try to check if I can produce that on an emulator (but not sure if it's possible because I've ever had a try before) when I get a spare time.
